src/CiviEntityStorage.php

```diff
@@ -258,7 +258,10 @@ public function countFieldData($storage_definition, $as_bool = FALSE) {
    * {@inheritdoc}
    */
   public function hasData() {
-    return $this->getCiviCrmApi()->getCount($this->entityTypeId, []) > 0;
+    if (!$this->civicrmApi) {
+      $this->civicrmApi = $this->getCiviCrmApi();
+    }
+    return $this->civicrmApi->getCount($this->entityType->get('civicrm_entity')) > 0;
```

Comment:
@KarinG Could you change this function to just:

```
return $this->getCiviCrmApi()->getCount($this->entityType->get('civicrm_entity')) > 0;
```

The `getCiviCrmApi()` already does the bit with caching the service in the class property, so there's no need to duplicate that.

But after, I think this is ready to merge! The issue with uninstall prompting you to delete all your Civi entities is a separate issue (which should get solved nonetheless, but we don't need to in this PR :-))

Reply:
Hi @dsnopek - as we discussed over a call - in D8.7 -> I need to: https://github.com/eileenmcnaughton/civicrm_entity/pull/184#discussion_r324813678 (items #2 and #3).

Reply:
I don't know how to solve that without simply checking 

`if (!$this->civicrmApi) {`

Reply:
@KarinG I'm not sure I understand. If the whole function is:

```
  public function hasData() {
    return $this->getCiviCrmApi()->getCount($this->entityType->get('civicrm_entity')) > 0;
  }
```

You shouldn't need to check `if (!$this->civicrmApi) {` because that's taken care of in `$this->getCiviCrmApi()`, there's no reason to duplicate that check here.

Or am I missing something?

Reply:
I don't understand this nearly as well as you do - but if I comment out the check:

```
  public function hasData() {
    //if (!$this->civicrmApi) {
    //  $this->civicrmApi = $this->getCiviCrmApi();
    ///}
    return $this->civicrmApi->getCount($this->entityType->get('civicrm_entity')) > 0;
  }

```

Then 
`root@9f03c6ef5d47:/var/www/drupal/web/modules/contrib/civicrm_entity/src# drush cr`

**And hit:**
`/admin/modules/uninstall`

I get many (exactly one for every CiviCRM Entity under: `/admin/structure/civicrm-entity`)
`Error: Call to a member function getCount() on null in Drupal\civicrm_entity\CiviEntityStorage->hasData() `

**Our env:**
Drupal 8.7.5
php -version: PHP 7.3.7 (cli) (built: Jul 17 2019 22:29:15) ( NTS )

Reply:
Hold on -> I see you have an additional () -> let me add that.

Reply:
Ok phew! Makes sense now. Sorry missed that first time around! Going to change the PR now.

Reply:
Thanks! Merging :-)